### PR TITLE
Take advantage of multi files COPY to simplify Dockerfile

### PR DESCRIPTION
**Pull request by elfosardo**
Move scripts under scripts dir and config files under config dir,
then copy everything at the same time when possible.
Also no need to specify ./ with COPY, the default path is always
inside the context of the build.

### CODE REVIEW

**Comment by elfosardo** [comment]
@hardys @maelk any chance I can get  review on this ?

File: Dockerfile [hunk 1]
```diff
@@ -33,7 +33,7 @@ FROM docker.io/centos:centos8
 
 ARG PKGS_LIST=main-packages-list.txt
 
-COPY ${PKGS_LIST} /tmp/main-packages-list.txt
+COPY ${PKGS_LIST} /tmp/
```
Comment (zaneb) [inline Dockerfile:36]: If the user supplies a different arg for PKGS_LIST we still want it to end up at /tmp/main-packages-list.txt, so I don't think this change is correct.

**Review by zaneb — COMMENTED** [review]
/lgtm cancel

File: Dockerfile (hunk 1 above)
Comment (elfosardo) [reply Dockerfile:36]: whoops this was not supposed to be here!

**Comment by elfosardo** [comment]
@zaneb hi! Can you please have another look? Thanks!

**Comment by elfosardo** [comment]
@zaneb @hardys @maelk can you please give this another look? Thanks!

**Comment by elfosardo** [comment]
@zaneb @hardys @maelk can you please give this another look? Thanks!

File: Dockerfile [hunk 2]
```diff
@@ -45,29 +45,17 @@ COPY --from=builder /tmp/ipxe/src/bin/undionly.kpxe /tmp/ipxe/src/bin-x86_64-efi
...
-# TODO(dtantsur): remove this script when we stop supporting running both
-# API and conductor processes via one entry point.
-COPY ./runironic.sh /bin/runironic
-
-COPY ./dnsmasq.conf.j2 /etc/dnsmasq.conf.j2
-COPY ./inspector.ipxe.j2 /tmp/inspector.ipxe.j2
-COPY ./dualboot.ipxe /tmp/dualboot.ipxe
+# TODO(dtantsur): remove scripts/runironic script when we stop supporting
+# running both API and conductor processes via one entry point.
+COPY scripts/ /bin/
```
Comment (zaneb) [inline Dockerfile:52]: Should this be `scripts/*`? This is correct syntax for rsync, but I can't find any indication in the [Dockerfile docs](https://docs.docker.com/engine/reference/builder/#copy) that the COPY directives treats a source with a trailing slash differently, rsync-style.

Comment (elfosardo) [reply Dockerfile:52]: from [COPY Docker docs](https://docs.docker.com/engine/reference/builder/#copy):
"If <src> is a directory, the entire contents of the directory are copied, including filesystem metadata."

Comment (zaneb) [reply Dockerfile:52]: And right after that:

>     Note
> 
>     The directory itself is not copied, just its contents.

Comment (zaneb) [reply Dockerfile:52]: You know I read that like 5 times, and each time my brain transposed the position of the `not` :facepalm:

**Review by dtantsur — APPROVED** [review]

**Pull request merged**